vendor/bat-native-ads/src/bat/ads/internal/notifications.cc

```diff
@@ -49,76 +49,91 @@ void Notifications::Initialize(InitializeCallback callback) {
 bool Notifications::Get(const std::string& id, NotificationInfo* info) const {
   DCHECK(is_initialized_);
 
-  auto notification = notifications_.find(id);
-  if (notification == notifications_.end()) {
+  auto iter = std::find_if(notifications_.begin(), notifications_.end(),
+      [&id](const auto& notification){return notification.id == id;});
+  if (iter == notifications_.end()) {
     return false;
   }
 
-  *info = notification->second;
+  *info = *iter;
 
   return true;
 }
 
-void Notifications::Add(const NotificationInfo& info) {
+void Notifications::PushBack(const NotificationInfo& info) {
   DCHECK(is_initialized_);
 
-  notifications_.insert({info.id, info});
+  notifications_.push_back(info);
+  ads_client_->ShowNotification(std::make_unique<NotificationInfo>(info));
 
   SaveState();
 }
 
-bool Notifications::Remove(const std::string& id) {
+void Notifications::PopFront(bool should_dismiss) {
+  if (!notifications_.empty()) {
```

Comment:
Please fail early and return if empty to save nesting code

Reply:
NONCRITICAL
Covered by: brave/brave-browser#6235 (1)